### PR TITLE
Removes sn-public-ip and combines into --service-node

### PR DESCRIPTION
**Pull request by Doy-lee**
Spell out service node for any end user interaction with the daemon for consistency. This lead to removing the --sn-public-ip because why not include it with the --service-node flag instead of a separate option.

Port kept separate for implementation ease

### CODE REVIEW

**Comment by msgmaxim** [comment]
Remove `--service-node` and add `--service-node-public-ip`? We already had `--service-node`, wouldn't it be confusing to change its meaning to something else?

**Comment by Doy-lee** [comment]
Seems superfluous to go --service-node, launch and it fails because you need the --service-node-public-ip and --storage-server-port. Combining it reduces the number of required args by 1.

Now it's just --service-node <ip> --storage-server-port <port>, still fails if you just specify --service-node <ip> which isn't nice but in interest of time I left it separate.

**Comment by Doy-lee** [comment]
So, instead of combining this just renames sn-public-ip to service-node-public-ip so there's no ambiguity and confusion about making the --service-node argument accept an ip as an argument.

File: src/cryptonote_core/cryptonote_core.cpp [hunk 1]
```diff
@@ -172,7 +172,7 @@ namespace cryptonote
   , "Run as a service node, options 'sn-public-ip' and 'storage-server-port' must be set"
```
Comment (msgmaxim) [inline src/cryptonote_core/cryptonote_core.cpp]: `service-node-public-ip` now

Comment (Doy-lee) [reply src/cryptonote_core/cryptonote_core.cpp]: Updated

**Pull request merged**